### DISCUSSION TITLE
How to disable rootless mode

### DISCUSSION THREAD

**Discussion by icefery** [containers/podman#23714] [Q&A] [answered]
I want to use podman with rootful mode and call `podman` command with normal user and without `sudo`.

This is docker doc example: https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user

> When the Docker daemon starts, it creates a Unix socket accessible by members of the docker group.

How to behave like this with podman?

Thanks

**Comment by rhatdan** [comment]
None of that is required.  Podman is designed to run as rootless, so just run it.  Only issue would be if the user running does not have entries in /etc/subuid and /etc/subgid.  But these entries should be added automatically via useradd command.

To run podman as root, run sudo podman, to run rootless run podman.

**Reply by rhatdan** [reply]
I wrote a blog on why doing this with Docker is a BAD idea, many years ago.

https://projectatomic.io/blog/2015/08/why-we-dont-let-non-root-users-run-docker-in-centos-fedora-or-rhel/

We created Podman to improve the security situation of Docker.

**Reply by icefery** [reply]
Blog is great!

Actually, I meet a problem, I started a container, it exited after I(user) exited.

I have searched this, it might be caused by zsh, but when I called `bash -l`, it still happened.

Is there any solution?

**Reply by rhatdan** [reply, marked as answer]
systemd kills all processes left in a session after a user exits.

https://github.com/containers/podman/blob/main/troubleshooting.md#17-rootless-containers-exit-once-the-user-session-exits

**Reply by icefery** [reply]
Thanks
